.travis.yml

```diff
@@ -37,8 +37,6 @@ addons:
 
 # Install various dependencies
 install:
-  - pip install --upgrade pytest
-  - pip install --upgrade pytest-cov
```

Comment:
> we'll need to update externals from time to time not to get stuck with old versions of the software

External packages don't have releases that frequently. The alternative is that everyone has a different version, including an older version that may or may not support some of the features we use. The alternative is also that every other user has to update these packages instead of a single user updating Spack once. In fact, I would like to go even further and say that Spack externals should include flake8 and sphinx as well. At least until we start releasing on PyPi and can specify these things in a `setup.py`.

> In any case I would leave the two lines above. This will let us work always with the latest release, and will expose any incompatibility with it as early as possible.

I agree with this. In https://github.com/LLNL/spack/pull/2659#issuecomment-268858229 I discovered something that `spack flake8` on my local computer didn't catch but Travis did. Presumably because Travis was using a newer version of flake8.